dex/src/main/scala/com/wavesplatform/dex/cli/Actions.scala

```diff
@@ -54,6 +56,59 @@ object Actions {
 
   private val backend = HttpURLConnectionBackend()
 
+  // noinspection ScalaStyle
+  def checkOrdersAgainstTxV3(args: Args, matcherSettings: MatcherSettings) = {
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    def checkPrice(price: Long, amountDecimals: Int, priceDecimals: Int): Try[Long] =
+      Try {
+        (BigDecimal(price) / BigDecimal(10).pow(priceDecimals - amountDecimals)).toBigInt.bigInteger.longValueExact()
+      }
+
+    def getAssetDecimalsOpt(assetsDb: AssetsDb[Id], asset: Asset): Option[Int] =
+      assetsDb.get(asset) match {
+        case Some(value) => value.decimals.some
+        case None =>
+          println(s"Error! Couldn't find asset decimals for asset $asset!")
+          None
+      }
+
+    cli.log(
+      s"""
+         |Passed arguments:
+         |  DEX config path : ${args.configPath}
+         |""".stripMargin
+    )
+
+    withLevelDb(matcherSettings.dataDirectory) { db =>
+      val apdb = AssetPairsDb.levelDb(db)
+      val obdb = OrderBookSnapshotDb.levelDb(db)
+      val apb = new AssetPairBuilder(matcherSettings, null, matcherSettings.blacklistedAssets)
+      val assetsDB = AssetsDb.levelDb(db)
+
+      val pairs = apdb.all()
+      val validPairs = pairs.flatMap(apb.quickValidateAssetPair(_).toOption)
+      validPairs.foreach { validPair =>
+        val result =
+          for {
+            (_, snapshot) <- obdb.get(validPair)
+            amountAssetDecimals <- getAssetDecimalsOpt(assetsDB, validPair.amountAsset)
+            priceAssetDecimals <- getAssetDecimalsOpt(assetsDB, validPair.priceAsset)
+          } yield (snapshot.bids.values ++ snapshot.asks.values).flatten.foreach { limitOrder =>
+            checkPrice(limitOrder.price, amountAssetDecimals, priceAssetDecimals) match {
+              case Failure(_) => println(s"pair: ${validPair.key}  oId: ${limitOrder.order.id()}  ts: ${limitOrder.order.timestamp}")
+              case Success(_) => // do nothing
```

Comment:
we can just

```
case _ =>
```

also, we can omit the comment cuz it's pretty straightforward logic in this particular place